### PR TITLE
:sparkles: feat(BaseMap): add harbors

### PR DESCRIPTION
**Pull request by regazzoj**
Ajout des ports sur la carte

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -43,6 +44,7 @@
         "eslint-plugin-promise": "^4.2.1",
         "eslint-plugin-standard": "^4.0.0",
         "eslint-plugin-vue": "^7.20.0",
+        "husky": "^4.3.8",
```
Comment (NicolasGrosjean) [inline package-lock.json:47]: husky come back again ...

I think you can remove it to simplify package-lock.json and yarn.lock diff

File: src/Store.ts [hunk 2]
```diff
@@ -92,21 +94,24 @@ export const store = {
...
   },
 
   updateBasemap (index: number): void {
     this.state.baseMap.setCurrentBasemap(index)
   },
 
   updateMap (): void {
-    this.state.baseMap.update(this.state.timeFilteredData)
+    this.state.baseMap.removeMarkers()
+    this.state.baseMap.addHarbors(this.harbors)
```
Comment (NicolasGrosjean) [inline src/Store.ts]: Harbors are removed and added again when dates are updated whereas they are independent from the dates

**Review by NicolasGrosjean — CHANGES_REQUESTED** [review]
The website is slower than before when ports are displayed.
I think we should reduce the number of ports to display, for example keep only the Mediterrannean ones or even keep only Italian, Tunisian and Lybian ports.

File: src/Store.ts (hunk 2 above)
Comment (regazzoj) [reply src/Store.ts]: Yep ! I would love to know how to create a layer for harbors, but I don't how to do it with MapBoxGlJS. Any idea ?

Comment (NicolasGrosjean) [reply src/Store.ts:113]: I found this : https://docs.mapbox.com/mapbox.js/example/v1.0.0/non-exclusive-markers/

Comment (NicolasGrosjean) [reply src/Store.ts]: In BaseMap.ts, we store all markers in `this.markers`, but we can store the harbors in another variable to manager them.

File: package-lock.json (hunk 1 above)
Comment (regazzoj) [reply package-lock.json:47]: Husky is used for pre-commit git hook. It runs lint-staged to make some checks

**Review by NicolasGrosjean — CHANGES_REQUESTED** [review]
`displayHarbors` continued to be called at each map update (when changing filtering date or toggle operative data slug). It is now fixed in my commit, but feel free to edit my commit.

The website is still slower than before when ports are displayed.
I think we should reduce the number of ports to display, for example keep only the Mediterrannean ones or even keep only Italian, Tunisian and Lybian ports.

**Review by NicolasGrosjean — CHANGES_REQUESTED** [review]
For me the code is very good.

Please find 2 suggestions which filter the port files, to avoid bad performance when moving on the map when harbors are displayed.
The files are in .txt because GitHub don't allow it in comment, but you need to change the extension to use them.
[ports_med.txt](https://github.com/CartONG/SOSMedDashboard/files/9053083/ports_med.txt)
[ports_small.txt](https://github.com/CartONG/SOSMedDashboard/files/9053084/ports_small.txt)

**Review by NicolasGrosjean — APPROVED** [review]
I don't see any more performance issue.

My last suggestion is to delete `src/assets/resources/harbors.json` like it is no more used, and we can find it again with Git if necessary.
There is no need to ask my review for this.

**Pull request merged**